File: 2.Add Two Numbers/Solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        head = ListNode(0)
        add, cur = 0, head

        while l1 or l2:
            _sum = add

            if l1:
                _sum += l1.val
                l1 = l1.next

            if l2:
                _sum += l2.val
                l2 = l2.next
            add, _sum = _sum / 10, _sum % 10
            cur.next = ListNode(_sum)
            cur = cur.next

        if add > 0:
            cur.next = ListNode(add)
            
        return head.next

File: 2.Add Two Numbers/Solution.go
type ListNode struct {
    Val int
    Next *ListNode
}
 
func addTwoNumbers(l1 *ListNode, l2 *ListNode) *ListNode {
    var (
        add int
        sum int
        cur *ListNode
    )
    head := ListNode{0, nil}
    add, sum, cur = 0, 0, &head
    for l1 != nil || l2 !=nil {
        sum = add
        if l1 != nil {
            sum += l1.Val
            l1 = l1.Next
        }
        if l2 != nil {
            sum += l2.Val
            l2 = l2.Next
        }
        add = sum / 10;
        sum = sum % 10;
        cur.Next = &ListNode{sum, nil}
        cur = cur.Next
    }
    if add > 0 {
        cur.Next = &ListNode{add, nil}
    }
    return head.Next
}